### PR TITLE
fix: Add protocol HTTPS configuration

### PR DESCRIPTION
**Pull request by loujaybee**
## Description

Adds changelog and docs for HTTPS protocol configuration. 

Related PR: https://github.com/gitpod-io/gitpod/pull/16547

## Related Issue(s)

N/A

## How to test

- View changelog
- View ports page
- View gitpod.yml page
- View gp cli page

## Release QA Checklist ✔️

<details>
  <summary>QA Checklist </summary>

## [Priority: High] Functional Testing

- [ ] Test all links to make sure they are working correctly
- [ ] Verify that all forms (sales, support, docs feedback)  are functioning properly and data is being captured correctly
- [ ] Check all buttons and calls-to-action to ensure they are leading to the correct pages
- [ ] Ensure that the website is mobile responsive and functions properly on different devices
- [ ] Ensure that page is optimal for Light/Dark Mode. ⇒ Text/Media Changes

## [Priority: High] Content Testing

- [ ] Check all website copies for accuracy and spelling/grammar errors
- [ ] Verify that all images and videos are displaying properly
- [ ] Ensure that all downloadable files are working correctly
- [ ] Check that all external links are functioning properly

## [Priority: Medium] Performance Testing

- [ ] Try to use optimal media assets. ([Decision Record - How To](https://github.com/gitpod-io/website/blob/main/docs/adr/0003-optimal-assets-format.md))
- [ ] Test website load times to ensure that the website is loading quickly
- [ ] Verify that the website can handle high traffic volumes without crashing
- [ ] Check that all website functionality is working properly under heavy load
- [ ] Perform cross-browser testing to ensure that the website works properly on all major browsers (Goolge Chrome, Firefox, Edge)

## [Priority: Medium] Accessibility Testing

- [ ] Ensure that the website meets basic accessibility guidelines.
- [ ] Test that users with disabilities can easily navigate and use the website
- [ ] Verify that all website content can be accessed using assistive technologies

By including these key items in your website release QA checklist, you can ensure that your website is fully functional, user-friendly, and meets your business requirements.

## Security Testing

> **Note**:💡 If we are changing something related to cookies or data storing or analytics side.

- [ ] Verify that the website is HTTPS secure and that there are no security vulnerabilities
- [ ] Ensure that all sensitive data is being stored and transmitted securely

  </details>

<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/3760"></a>

### CODE REVIEW

File: src/lib/contents/changelog/https-ports.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+<script context="module">
+  export const prerender = true;
+</script>
+
+In Gitpod, you can now expose ports from your workspace running with HTTPS.
+
+Previously, HTTPS ports exposed in a workspace would show a "port not found" page when accessed. You can configure the port protocol either in your [gitpod.yml](/docs/references/gitpod-yml) or via the [Gitpod CLI](/docs/references/gitpod-cli).
+
+## Configuring a port protocol in `.giptod.yml`
```
Comment (laushinka) [inline src/lib/contents/changelog/https-ports.md:19]:
```suggestion
## Configuring a port protocol in `.gitpod.yml`
```

File: gitpod/docs/references/gitpod-cli.md [hunk 2]
```diff
@@ -318,6 +318,22 @@ Here's an example which will make port `3000` private:
 gp ports visibility 3000:private
 ```
 
+### protocol
+
+Be default ports running in the workspace are assumed to be HTTP. You can dynamically configure (without a workspace restart) a port to use HTTPS with the `gp ports protocol` command.
```
Comment (laushinka) [inline gitpod/docs/references/gitpod-cli.md:323]:
```suggestion
By default ports running in the workspace are assumed to be HTTP. You can dynamically configure (without a workspace restart) a port to use HTTPS with the `gp ports protocol` command.
```

File: gitpod/docs/configure/workspaces/ports.md [hunk 3]
```diff
@@ -242,3 +242,35 @@ To make this work, your web application's `fetch` request needs to have the `cre
 **Configure your server**
 
 In your server (the one on port 5001 in the above example), you have to configure the response to include the `Access-Control-Allow-Credentials` header. Without it, your browser rejects the response and you see CORS errors in the browser console.
+
+## Port protocols
+
+Be default ports running in the workspace are assumed to be HTTP. You can configure your port to use HTTPS by updating the `.gitpod.yml` or using the `gp ports protocol` command.
```
Comment (laushinka) [inline gitpod/docs/configure/workspaces/ports.md:248]:
```suggestion
By default ports running in the workspace are assumed to be HTTP. You can configure your port to use HTTPS by updating the `.gitpod.yml` or using the `gp ports protocol` command.
```

**Comment by loujaybee** [comment]
Great catches, thanks @laushinka !

**Comment by Siddhant-K-code** [comment]
Updated the branch due to a critical 🐛 fix (observed due to this PR, so thanks @loujaybee) in #3761

**Comment by loujaybee** [comment]
@laushinka, @akosyakov, @iQQBot, @Siddhant-K-code -> Any chance I could get a review + approval if you're happy? 

Keen to ship this ✅

File: src/lib/contents/changelog/https-ports.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
+---
+title: Workspaces forward ports using the HTTPS protocol
+alt: An example gitpod.yml configuration showing the protocol property set to https
+date: 2023-06-16
```
Comment (loujaybee) [inline src/lib/contents/changelog/https-ports.md:4]:
```suggestion
date: 2023-06-19
```

**Pull request merged**